src/openforms/formio/dynamic_config/dynamic_options.py

```diff
@@ -0,0 +1,56 @@
+from django.template.defaultfilters import escape_filter as escape
+
+from glom import assign, glom
+from json_logic import jsonLogic
+
+from openforms.logging import logevent
+from openforms.submissions.models import Submission
+from openforms.typing import DataMapping
+
+from ..typing import Component
+
+
+def add_options_to_config(
+    component: Component,
+    data: DataMapping,
+    submission: Submission,
+    options_path: str = "values",
+    data_src_path: str = "dataSrc",
+) -> None:
+    if glom(component, data_src_path, default=None) != "variable":
+        return
+
+    items_path = glom(component, "data.itemsExpression")
+
+    # The array of items from which we need to get the values
+    items_array = jsonLogic(items_path, data)
+
+    if not items_array:
+        return
+
+    # Is each item a dict, like for repeating groups, or are they primitives
+    # ready to use?
+    is_obj = isinstance(items_array[0], dict)
```

Comment:
what happens if they're not a homogenous type?

Reply:
Then we ask for that hypothesis lightning talk.

Reply:
So far we have the two cases:
1. It's just a string, so there is no need to specify `valueExpression` in the configuration of the component
2. It's an object, so we use `valueExpression` to get the value for the object. 
But if they are different types, we have no way to configure 'for option 1, make the value in way X, for option 2 in way Y.' So for now, non-homogeneous types are not expected in this PR.

Although I should probably add the check that the variable configured is an array :thinking:

Reply:
Created follow up tickets in #2623 and #2624